pkg/azure/externalmetrics/monitor_client.go

```diff
@@ -55,23 +57,42 @@ func (c *monitorClient) GetAzureMetric(azMetricRequest AzureExternalMetricReques
 		return AzureExternalMetricResponse{}, err
 	}
 
-	total := extractValue(metricResult)
-
-	klog.V(2).Infof("found metric value: %f", total)
+	value, err := extractValue(azMetricRequest, metricResult)
 
 	// TODO set Value based on aggregations type
 	return AzureExternalMetricResponse{
-		Total: total,
-	}, nil
+		Value: value,
+	}, err
 }
 
-func extractValue(metricResult insights.Response) float64 {
-	//TODO extract value based on aggregation type
-	//TODO check for nils
+func extractValue(azMetricRequest AzureExternalMetricRequest, metricResult insights.Response) (float64, error) {
 	metricVals := *metricResult.Value
 	Timeseries := *metricVals[0].Timeseries
 	data := *Timeseries[0].Data
-	total := *data[len(data)-1].Total
```

Comment:
It wasn't in here before but can we add nil checks for each pointer pull off the struct in the case some bad data comes back?